interop/src/clients/corecrypto/ios.rs

```diff
@@ -0,0 +1,367 @@
+            ])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("Failed to launch application");
+
+        let mut output = BufReader::new(
+            process
+                .stdout
+                .take()
+                .expect("Expected stdout to be available on child process"),
+        );
+        let mut line = String::new();
+
+        // Waiting for confirmation that the application has launched.
+        while !line.contains("Ready") {
+            line.clear();
+            output
+                .read_line(&mut line)
+                .expect("was expecting ready signal on stdout");
+        }
+
+        log::info!("application launched: {}", line);
+
+        Self {
+            device,
+            process,
+            output: Rc::new(RefCell::new(output)),
+        }
+    }
+
+    async fn execute(&self, action: String) -> Result<String> {
+        log::info!("interop://{}", action);
+
+        Command::new("xcrun")
+            .args([
+                "simctl",
+                "openurl",
+                &self.device,
+                format!("interop://{}", action).as_str(),
+            ])
+            .output()
+            .expect("Failed to execute action");
+
+        let mut result = String::new();
+        let mut output = self.output.try_borrow_mut()?;
+
+        output.read_line(&mut result)?;
+
+        log::info!("{}", result);
+
+        let result: InteropResult = serde_json::from_str(result.trim())?;
+
+        match result {
+            InteropResult::Success { value } => Ok(value),
+            InteropResult::Failure { message } => Err(SimulatorDriverError { msg: message }.into()),
+        }
+    }
+}
+
+impl Drop for SimulatorDriver {
+    fn drop(&mut self) {
+        self.process.kill().expect("expected child process to be killed")
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct CoreCryptoIosClient {
+    driver: SimulatorDriver,
+    client_id: Vec<u8>,
+    #[cfg(feature = "proteus")]
+    prekey_last_id: Cell<u16>,
+}
+
+impl CoreCryptoIosClient {
+    pub(crate) async fn new() -> Result<Self> {
+        let client_id = uuid::Uuid::new_v4();
+        let client_id_str = client_id.as_hyphenated().to_string();
+        let client_id_base64 = general_purpose::STANDARD.encode(client_id_str.as_str());
+        let ciphersuite = CIPHERSUITE_IN_USE as u16;
+
+        let driver = SimulatorDriver::new("booted".into(), "com.wire.InteropClient".into());
+        log::info!("initialising core crypto with ciphersuite {}", ciphersuite);
+        driver
+            .execute(format!(
+                "init-mls?client={}&ciphersuite={}",
+                client_id_base64, ciphersuite
+            ))
+            .await?;
+
+        Ok(Self {
+            driver,
+            client_id: client_id.into_bytes().into(),
+            #[cfg(feature = "proteus")]
+            prekey_last_id: Cell::new(0),
+        })
+    }
+
+    #[allow(dead_code)]
```

Comment:
Why is this necessary?

Reply:
It isn't. Was implemented for the web client but we can actually delete it there as well.

Reply:
Deleted